netbox/utilities/tests/test_counters.py

```diff
@@ -79,3 +82,33 @@ def test_interface_count_move(self):
         device2.refresh_from_db()
         self.assertEqual(device1.interface_count, 1)
         self.assertEqual(device2.interface_count, 3)
+
+    @override_settings(EXEMPT_VIEW_PERMISSIONS=['*'])
+    def test_mptt_child_delete(self):
+        device1, device2 = Device.objects.all()
+        inventory_item1 = InventoryItem.objects.create(device=device1, name='Inventory Item 1')
+        inventory_item2 = InventoryItem.objects.create(device=device1, name='Inventory Item 1', parent=inventory_item1)
+        device1.refresh_from_db()
+        self.assertEqual(device1.inventory_item_count, 2)
+
+        # Setup bulk_delete for the inventory items
+        pk_list = device1.inventoryitems.values_list('pk', flat=True)
+        data = {
+            'pk': pk_list,
+            'confirm': True,
+            '_confirm': True,  # Form button
+        }
+
+        # Assign unconstrained permission
+        obj_perm = ObjectPermission(
+            name='Test permission',
+            actions=['delete']
+        )
+        obj_perm.save()
+        obj_perm.users.add(self.user.pk)
+        obj_perm.object_types.add(ContentType.objects.get_for_model(InventoryItem))
```

Comment:
You can use the `add_permissions()` test case method for this.
```suggestion
        self.add_permissions('dcim.delete_inventoryitem')
```